testing/logging/test_fixture.py

```diff
@@ -290,25 +294,47 @@ def test_caplog_captures_for_all_stages(caplog, logging_during_setup_and_teardow
     assert [x.message for x in caplog.get_records("setup")] == ["a_setup_log"]
 
     # This reaches into private API, don't use this type of thing in real tests!
-    assert set(caplog._item.stash[caplog_records_key]) == {"setup", "call"}
+    # error: Invalid index type "StashKey[dict[str, list[LogRecord]]]" for "Stash";
```

Comment:
I tried playing a bit with the `StashKey`'s `TypeVar` variance to fix this, but I *think* it needs to be invariant because it's used in both setters and getters. Maybe the new Python 3.12 type parameter syntax fixes this, didn't check.

Anyway I think it can be worked around by doing it in two lines:

```py
caplog_records = caplog._item.stash[caplog_records_key]
assert set(caplog_records) == {"setup", "call"}
```

Reply:
Indeed that fixes it, however I did not get why... could you explain it, please?